## Perguntas a serem respondidas.

1. Quantas casas estão disponíveis para compra?
2. Quantos atributos as casas possuem?
3. Quais são os atributos das casas?
4. Qual a casa mais cara ( casa com o maior valor de venda )?
5. Qual a casa com o maior número de quartos?
6. Qual a soma total de quartos do conjunto de dados?
7. Quantas casas possuem 2 banheiros?
8. Qual o preço médio de todas as casas no conjunto de dados?
9. Qual o preço médio de casas com 2 banheiros?
10. Qual o preço mínimo entre as casas com 3 quartos?
11. Quantas casas possuem mais de 300 metros quadrados na sala de estar?
12. Quantas casas tem mais de 2 andares?
13. Quantas casas tem vista para o mar?
14. Das casas com vista para o mar, quantas tem 3 quartos?
15. Das casas com mais de 300 metros quadrados de sala de estar, quantas tem mais de 2 banheiros?

In [23]:
import pandas as pd
import numpy as np

In [5]:
df = pd.read_csv('kc_house_data.csv')

## RESPOSTAS

### 1. Quantas casas estão disponíveis para compra?


In [6]:
df.shape # especificamente 21.613 casas

(21613, 21)

### 2. Quantos atributos as casas possuem?


De acordo com as colunas mostradas acima são 21 atributos

### 3. Quais são os atributos das casas?

In [7]:
df.columns

Index(['id', 'date', 'price', 'bedrooms', 'bathrooms', 'sqft_living',
       'sqft_lot', 'floors', 'waterfront', 'view', 'condition', 'grade',
       'sqft_above', 'sqft_basement', 'yr_built', 'yr_renovated', 'zipcode',
       'lat', 'long', 'sqft_living15', 'sqft_lot15'],
      dtype='object')

### 4. Qual a casa mais cara ( casa com o maior valor de venda )?


In [28]:
df['price'].sort_values(ascending=False).head(1)

7252    7700000.0
Name: price, dtype: float64

#### casa no valor de U$ 7.700.000,00 

### 5. Qual a casa com o maior número de quartos?


In [29]:
df['bedrooms'].sort_values(ascending=False).head(1)

15870    33
Name: bedrooms, dtype: int64

#### Casa com 33 quartos

### 6. Qual a soma total de quartos do conjunto de dados?


In [10]:
df['bedrooms'].sum()

72854

### 7. Quantas casas possuem 2 banheiros?


In [11]:
df['bathrooms'].value_counts()[2.00]

1930

#### 1930 casas com 2 banheiros

### 8. Qual o preço médio de todas as casas no conjunto de dados?


In [12]:
df['price'].mean()

540088.1417665294

### 9. Qual o preço médio de casas com 2 banheiros?


In [24]:
np.round(df.loc[df['bathrooms'] == 2, 'price'].mean(), 2)

457889.72

### 10. Qual o preço mínimo entre as casas com 3 quartos?


In [27]:
df.loc[df['bedrooms'] == 3, 'price'].min()

82000.0

### 11. Quantas casas possuem mais de 300 metros quadrados na sala de estar?


#### _Primeiro precisamos converter de pés pra metros quadrados_

In [63]:
df['living_m2'] = df['sqft_living'] * 0.093

In [64]:
df.loc[df['living_m2'] > 300].shape

(2258, 22)

#### _Um total de 2.258 casas_

### 12. Quantas casas tem mais de 2 andares?


In [41]:
df.loc[df['floors'] > 2].shape

(782, 21)

#### _Um total de 782 casas_

### 13. Quantas casas tem vista para o mar?


In [44]:
df.loc[df['waterfront'] >= 1].shape

(163, 21)

#### _Um total de 163 casas_

### 14. Das casas com vista para o mar, quantas tem 3 quartos?


In [55]:
houses_waterfront = df.loc[df['waterfront'] >= 1]
houses_waterfront.loc[houses_waterfront['bedrooms'] == 3].shape

(64, 21)

#### _Um total de 64 casas_

### 15. Das casas com mais de 300 metros quadrados de sala de estar, quantas tem mais de 2 banheiros?

In [65]:
houses_more_than_300 = df.loc[df['living_m2'] > 300]

In [66]:
houses_more_than_300.loc[houses_more_than_300['bathrooms'] > 2].shape
#len(houses_more_than_300.loc[houses_more_than_300['bathrooms'] > 2]) --> Outra forma de fazer


(2201, 22)

#### _Um total de 2.201 casas_